modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -6181,13 +6184,30 @@ public String removeNonNameElementToCamelCase(String name) {
      * @return camelized string
      */
     protected String removeNonNameElementToCamelCase(final String name, final String nonNameElementPattern) {
-        String result = Arrays.stream(name.split(nonNameElementPattern))
-                .map(StringUtils::capitalize)
-                .collect(Collectors.joining(""));
-        if (result.length() > 0) {
-            result = result.substring(0, 1).toLowerCase(Locale.ROOT) + result.substring(1);
+        if (Boolean.parseBoolean(System.getProperty("openapi.generator.fix.camelize"))) {
+            // new behaviour with fix
+            String[] splitString = name.split(nonNameElementPattern);
+
+            if (splitString.length > 0) {
+                splitString[0] = camelize(splitString[0], CamelizeOption.LOWERCASE_FIRST_CHAR);
+            }
+
+            String result = Arrays.stream(splitString)
+                    .map(StringUtils::capitalize)
```

Comment:
Why would we capitalize splitString[0] when we just used camelize with lower case first?